app/src/main/java/com/example/drawingApp/MainActivity.kt

```diff
@@ -1,23 +1,111 @@
 package com.example.drawingApp
 
 import android.os.Bundle
+import android.os.StrictMode
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.EditText
+import android.widget.TextView
+import android.widget.Toast
 import androidx.activity.viewModels
+import androidx.appcompat.app.AlertDialog
 import androidx.appcompat.app.AppCompatActivity
+import androidx.core.widget.addTextChangedListener
 
 class MainActivity : AppCompatActivity() {
+    private val MAX_CHAR_LEN = 16
     private val model: MainViewModel by viewModels()
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
-
+        enableStrictMode()
         val colorCircleView = findViewById<SelectedColorView>(R.id.colorCircle)
         val drawingFieldView = findViewById<DrawingFieldView>(R.id.drawField)
 
+        drawingFieldView.onFinishLine = {
+            model.activeBitmap = it
+        }
+        findViewById<TextView>(R.id.newLayer).setOnClickListener {
+            showLayerAlertDialog()
+        }
+
+        model.activeBitmap?.let {
+            drawingFieldView.setBitmap(it)
+            drawingFieldView.invalidate()
+        }
+
+
+
         colorCircleView.onColorChange = { stateColor ->
             model.primaryColor = stateColor
             drawingFieldView.updateColor(stateColor)
         }
 
+        if (model.isDialogOpen) showLayerAlertDialog()
+
+    }
+
+    fun showLayerAlertDialog() {
+        val inflater = LayoutInflater.from(this@MainActivity)
+        val view = inflater.inflate(R.layout.layer_dialog, null)
+        val layerField = view.findViewById<EditText>(R.id.newLayerField)
+        val errorText = view.findViewById<TextView>(R.id.dialogCharCountError)
+        var layerCounter = 1
+
+        //find the first available layer name not taken
+        while (model.layerNames.contains("Layer ${layerCounter}")) {
+            layerCounter += 1
+        }
+        layerField.setText(getString(R.string.new_layer_hint, layerCounter))
+
+
+        //needs to be after the set text or else the textchanged listener will overwrite the previous text needed for rotations
+        layerField.addTextChangedListener {
+            model.currentDialogText = layerField.text.toString()
+            if (it.toString().trim().length > MAX_CHAR_LEN) {
+                errorText.visibility = View.VISIBLE
```

Comment:
`errorText.isVisible = it.toString().trim().length > MAX_CHAR_LEN`?

Reply:
implemented but it feels far more awkward than the previous.

Reply:
No reason to use an if statement when you are already processing a boolean result